### DISCUSSION TITLE
Please consider creating `langium-core` package to support parse only usage

### DISCUSSION THREAD

**Discussion by brsanthu** [eclipse-langium/langium#1489] [Q&A]
We are working on serverside only parser based on langium. In our package, we removed all dependencies except on `langium`, still the package is close to 3mb I guess it is due to langium bring in `vscode-languageserver` which is not required for parse only usage.

Can you please consider creating `langium-core` with parser only related dependencies, which `langium` will depend on? This way, any users who needs parser only, they can include `langium-core` to reduce the size and dependencies.

**Comment by msujew** [comment]
Hey @brsanthu,

as of v3.0 of Langium, just importing `langium` (and not `langium/lsp`) will bring you the same benefits. We specifically split those imports to reduce bundle size. The `langium` import has no conpile time dependency on `vscode-languageserver`.
